dirs.proj

```diff
@@ -5,7 +5,9 @@
      <Solution Include="Microsoft.ApplicationInsights.sln" />    
   </ItemGroup>
   <Target Name="Build">
-    <MSBuild Projects="@(Solution)" Targets="Restore;Build" ContinueOnError="ErrorAndStop" Properties="PreReleaseVersion=$(PreReleaseVersion)"/> 
+	<Exec Command='IF EXIST "%(Solution.FullPath)" nuget.exe restore "%(Solution.FullPath)" -NonInteractive' 
```

Comment:
curious why not do it as a build step?

Reply:
it everything is done via dirs.proj, then doing local runs mocks the actual behaviour in build servers.